receiver/skywalkingreceiver/factory.go

```
@@ -0,0 +1,114 @@
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+const (
+	typeStr = "skywalking"
+
+	// Protocol values.
+	protoGRPC = "grpc"
+
+	// Default endpoints to bind to.
+	defaultGRPCBindEndpoint = "0.0.0.0:11800"
+	defaultHTTPBindEndpoint = "0.0.0.0:12800"
+)
+
+// NewFactory creates a new Jaeger receiver factory.
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiverhelper.WithTraces(createTracesReceiver))
+}
+
+// CreateDefaultConfig creates the default configuration for Skywalking receiver.
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ReceiverSettings: config.NewReceiverSettings(config.NewComponentID(typeStr)),
+		Protocols: Protocols{
+			GRPC: &configgrpc.GRPCServerSettings{
+				NetAddr: confignet.NetAddr{
+					Endpoint:  defaultGRPCBindEndpoint,
+					Transport: "tcp",
+				},
+			},
+			HTTP: &confighttp.HTTPServerSettings{
+				Endpoint: defaultHTTPBindEndpoint,
+			},
+		},
+	}
+}
+
+// createTracesReceiver creates a trace receiver based on provided config.
+func createTracesReceiver(
+	_ context.Context,
+	set component.ReceiverCreateSettings,
+	cfg config.Receiver,
+	nextConsumer consumer.Traces,
+) (component.TracesReceiver, error) {
+
+	// Convert settings in the source config to configuration struct
+	// that Skywalking receiver understands.
+	rCfg := cfg.(*Config)
+
+	var config configuration
+	// Set ports
+	if rCfg.Protocols.GRPC != nil {
+		config.CollectorGRPCServerSettings = *rCfg.Protocols.GRPC
+		config.CollectorGRPCPort, _ = extractPortFromEndpoint(rCfg.Protocols.GRPC.NetAddr.Endpoint)
+	}
+
+	if rCfg.Protocols.HTTP != nil {
+		config.CollectorHTTPSettings = *rCfg.Protocols.HTTP
+		config.CollectorHTTPPort, _ = extractPortFromEndpoint(rCfg.Protocols.HTTP.Endpoint)
```

Similarly as above, error is not checked here. Should it return `nil, err` in case `extractPortFromEndpoint` fails?